packages/auto-approve/src/utils-for-PR-checking.ts

```diff
@@ -0,0 +1,188 @@
+): {file: File; fileRule: FileSpecificRule} | undefined {
+  let file;
+  let fileRule;
+
+  for (const i in changedFiles) {
+    for (const j in languageRules) {
+      if (
+        languageRules[j].prAuthor === author &&
+        languageRules[j].targetFile === changedFiles[i].filename
+      ) {
+        file = changedFiles[i];
+        fileRule = languageRules[j];
+        break;
+      }
+    }
+  }
+
+  // If we didn't find a match, return undefined, but this shouldn't
+  // stop execution of the rest of the tests
+  if (!(file && fileRule)) {
+    return undefined;
+  } else {
+    return {file, fileRule};
+  }
+}
+
+/**
+ * Given a patch for a file that was changed in a PR, and a regular expression to search
+ * for the old version number and a regular expression to search for the new version number,
+ * this function will return the old and new versions of a package.
+ *
+ * @param versionFile the changed file that has additional rules to conform to
+ * @param oldVersionRegex the regular exp to find the old version number of whatever is being changed
+ * @param newVersionRegex the regular exp to find the new version number of whatever is being changed
+ * @returns the previous and new major and minor versions of a package in an object containing those 4 properties.
+ */
+export function getVersions(
+  versionFile: File | undefined,
+  oldVersionRegex: string,
+  newVersionRegex: string
+): Versions | undefined {
+  if (!versionFile) {
+    return undefined;
+  }
+
+  let oldMajorVersion;
+  let oldMinorVersion;
+  let newMajorVersion;
+  let newMinorVersion;
+
+  const oldVersions = versionFile.patch?.match(new RegExp(oldVersionRegex));
+  const newVersions = versionFile.patch?.match(new RegExp(newVersionRegex));
+
+  if (oldVersions) {
+    oldMajorVersion = oldVersions[1];
+    oldMinorVersion = oldVersions[2];
+  }
+
+  if (newVersions) {
+    newMajorVersion = newVersions[1];
+    newMinorVersion = newVersions[2];
+  }
+
+  // If there is a change with a file that requires special validation checks,
+  // and we can't find these pieces of information, we should throw an error, and not
+  // perform any other checks, since that would open us up to potentially merging a
+  // sensitive file without having proper checks.
+  if (
+    !(oldMajorVersion && oldMinorVersion && newMajorVersion && newMinorVersion)
+  ) {
+    throw Error(
+      `Could not find versions in ${versionFile.filename}/${versionFile.sha}`
+    );
+  }
+  return {oldMajorVersion, oldMinorVersion, newMajorVersion, newMinorVersion};
+}
+
+/**
+ * This function determines whether the major version of a package was changed.
+ *
+ * @param versions an object containing the previous and newer versions of the package being updated
+ * @returns whether the major version changed.
+ */
+export function isMajorVersionChanging(versions: Versions): boolean {
+  return Number(versions.newMajorVersion) - Number(versions.oldMajorVersion) ===
+    0
+    ? false
+    : true;
+}
```

Comment:
Minor, but this may be simplified to:

```suggestion
export function isMajorVersionChanging(versions: Versions): boolean {
  return versions.newMajorVersion === versions.oldMajorVersion;
}
```